src/main/javascript/directives/cat-confirm-click.js

```diff
@@ -3,18 +3,34 @@
 /**
  * @ngdoc directive
  * @name cat.directives.confirmClick:catConfirmClick
+ * 
+ * @param cat-i18n-params cat-i18n-params need to be parsed as JSON or it will not work
  */
 angular.module('cat.directives.confirmClick', [])
-    .directive('catConfirmClick', function CatConfirmClickDirective() {
+    .directive('catConfirmClick', function CatConfirmClickDirective(catI18nService) {
         return {
             restrict: 'A',
             link: function CatConfirmClickLink(scope, element, attr) {
                 var msg = attr.catConfirmClick || 'Are you sure?';
                 var clickAction = attr.catOnConfirm;
+                var params = undefined;
+                if (!!attr.catI18nParams) {
+                    params = JSON.parse(attr.catI18nParams);
```

Comment:
i think what you want to do is parse it as an angular expression
see https://docs.angularjs.org/api/ng/service/$parse